File: cmux-tui/bindings/java/src/com/cmux/TopologyOperation.java
package com.cmux;

public enum TopologyOperation {
    WORKSPACE_CREATED("workspace-created"), SCREEN_CREATED("screen-created"),
    PANE_SPLIT("pane-split"), SURFACE_ATTACHED("surface-attached"),
    SURFACE_CLOSED("surface-closed"), PANE_CLOSED("pane-closed"),
    SCREEN_CLOSED("screen-closed"), WORKSPACE_CLOSED("workspace-closed"),
    WORKSPACE_RENAMED("workspace-renamed"), SCREEN_RENAMED("screen-renamed"),
    PANE_RENAMED("pane-renamed"), SURFACE_RENAMED("surface-renamed"),
    SPLIT_RATIO_CHANGED("split-ratio-changed"), PANES_SWAPPED("panes-swapped"),
    LAYOUT_APPLIED("layout-applied"), TAB_MOVED("tab-moved"),
    WORKSPACE_MOVED("workspace-moved");

    private final String wire;
    TopologyOperation(String wire) { this.wire = wire; }
    public String wire() { return wire; }

    static TopologyOperation from(String wire) {
        for (TopologyOperation value : values()) if (value.wire.equals(wire)) return value;
        throw new IllegalArgumentException("invalid topology operation " + wire);
    }
}

File: Packages/macOS/CmuxTerminalBackend/Sources/CmuxTerminalBackend/TopologyOperation.swift
/// The committed structural operation described by a topology delta.
public enum TopologyOperation: String, Codable, CaseIterable, Sendable {
    /// A workspace was created.
    case workspaceCreated = "workspace-created"

    /// A screen was created.
    case screenCreated = "screen-created"

    /// A pane was split into a new layout branch.
    case paneSplit = "pane-split"

    /// A surface was attached to a pane.
    case surfaceAttached = "surface-attached"

    /// A surface runtime was atomically replaced under the same stable UUID.
    case surfaceReplaced = "surface-replaced"

    /// A surface was closed.
    case surfaceClosed = "surface-closed"

    /// A pane was closed.
    case paneClosed = "pane-closed"

    /// A screen was closed.
    case screenClosed = "screen-closed"

    /// A workspace was closed.
    case workspaceClosed = "workspace-closed"

    /// A workspace name changed.
    case workspaceRenamed = "workspace-renamed"

    /// A screen name changed.
    case screenRenamed = "screen-renamed"

    /// A pane name changed.
    case paneRenamed = "pane-renamed"

    /// A surface name changed.
    case surfaceRenamed = "surface-renamed"

    /// A split ratio changed.
    case splitRatioChanged = "split-ratio-changed"

    /// Two panes exchanged positions.
    case panesSwapped = "panes-swapped"

    /// A complete layout was applied.
    case layoutApplied = "layout-applied"

    /// A tab moved between positions or panes.
    case tabMoved = "tab-moved"

    /// A workspace moved within the canonical ordering.
    case workspaceMoved = "workspace-moved"
}
